hwy/tests/convert_test.cc

```diff
@@ -832,6 +832,227 @@ HWY_NOINLINE void TestAllFloatFromUint() {
+                        static_cast<TF>(IsSigned<TTo>() ? 1 : 2));
+    static_assert(kMinOutOfRangePosVal > static_cast<TF>(0),
+                  "kMinOutOfRangePosVal > 0 must be true");
+
+    TU scalar_non_elided_zero = static_cast<TU>(0);
+#if HWY_COMPILER_MSVC
+    scalar_non_elided_zero =
+        *static_cast<const volatile TU*>(&scalar_non_elided_zero);
+#else
+    PreventElision(scalar_non_elided_zero);
+#endif
+
+    const Rebind<TTo, DF> d_to;
+    const RebindToSigned<decltype(d_to)> di_to;
+    const RebindToUnsigned<DF> du;
+
+    const auto non_elided_zero = BitCast(df, Set(du, scalar_non_elided_zero));
+
+    const auto v2 = Or(non_elided_zero, v);
+    const auto is_nan_mask = IsNaN(v2);
+    const auto is_in_range_mask =
+        AndNot(is_nan_mask, Lt(Abs(IfThenZeroElse(is_nan_mask, v2)),
+                               Set(df, kMinOutOfRangePosVal)));
+
+    const auto is_nan_vmask = VecFromMask(d_to, DoF2IConvMask(df, is_nan_mask));
+
+    const auto expected_in_range =
+        DoF2IConvVec(df, IfThenElseZero(is_in_range_mask, v2));
+    const auto expected_out_of_range =
+        Or(is_nan_vmask,
+           BitCast(d_to, IfNegativeThenElse(
+                             DoF2IConvMsbMaskVec(df, v2),
+                             BitCast(di_to, Set(d_to, LimitsMin<TTo>())),
+                             BitCast(di_to, Set(d_to, LimitsMax<TTo>())))));
+
+    const auto expected = IfThenElse(DoF2IConvMask(df, is_in_range_mask),
+                                     expected_in_range, expected_out_of_range);
+
+    AssertVecEqual(d_to, expected, Or(DoF2IConvVec(df, v), is_nan_vmask),
+                   filename, line);
+    AssertVecEqual(d_to, expected, Or(DoF2IConvVec(df, v2), is_nan_vmask),
+                   filename, line);
+  }
+
+ public:
+  template <typename TF, class DF>
+  HWY_NOINLINE void operator()(TF /*unused*/, const DF df) {
+    using TI = MakeSigned<TF>;
+    using TU = MakeUnsigned<TF>;
+    const RebindToSigned<DF> di;
+
+    // TODO(janwas): workaround for QEMU 7.2 crash on vfwcvt_rtz_x_f_v:
+    // target/riscv/translate.c:213 in void decode_save_opc(DisasContext *):
+    // ctx->insn_start != NULL.
+#if HWY_TARGET == HWY_RVV || (HWY_ARCH_RVV && HWY_TARGET == HWY_EMU128)
+    if (sizeof(TTo) > sizeof(TF)) {
+      return;
+    }
+#endif
+
+    const auto pos_nan = BitCast(df, Set(di, LimitsMax<TI>()));
+    const auto neg_nan = BitCast(df, Set(di, static_cast<TI>(-1)));
+    const auto pos_inf =
+        BitCast(df, Set(di, static_cast<TI>(ExponentMask<TF>())));
+    const auto neg_inf = Neg(pos_inf);
+
+    VerifyNonFiniteF2I(df, pos_nan, __FILE__, __LINE__);
+    VerifyNonFiniteF2I(df, neg_nan, __FILE__, __LINE__);
+    VerifyNonFiniteF2I(df, pos_inf, __FILE__, __LINE__);
+    VerifyNonFiniteF2I(df, neg_inf, __FILE__, __LINE__);
+
+    TI non_elided_one = static_cast<TI>(1);
+#if HWY_COMPILER_MSVC
+    non_elided_one = *reinterpret_cast<const volatile TI*>(&non_elided_one);
```

Comment:
Here also Unpredictable1, and also below?

Reply:
Already replaced the above code with Unpredictable1()